src/auth/googleauth.ts

```diff
@@ -634,4 +643,59 @@ export class GoogleAuth {
           if (callback) callback(null, body);
         });
   }
+
+
+  /**
+   * Returns the contents of the metadata of GC instance
+   * @return object representation of the service account JSON
+   */
+  getCredentials(
+      callback?: (err: Error|null, credentials?: CredentialBody) => void) {
+    if (this.jsonContent) {
+      const credential: CredentialBody = {
+        client_email: this.jsonContent.client_email,
+        private_key: this.jsonContent.private_key
+      };
+      if (callback) {
+        callback(null, credential);
+      }
+    } else {
+      this._checkIsGCE((err, gce) => {
+        if (err) {
+          if (callback) {
+            callback(err, undefined);
+          }
+        } else if (gce) {
```

Comment:
nit: it would make sense to reverse this branch and move else up under `else if (!gce)`. That way error/unecpected cases are dealt with first.

Reply:
Done